proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1645 @@
+> 'directions' the interpolation could move, following different logical rules.
+
+This process accepts two hue angles (`hue1` and `hue2`), and returns both hues
+adjusted according to the given `method`. When no hue interpolation `method` is
+specified, the default is `shorter`.
+
+The process for each [hue interpolation method][hue-interpolation] is defined
+in [CSS Color Level 4][color-4]. If the `method` is not the value
+`'specified'`, both angles need to be constrained to `[0, 360)` prior to
+interpolation.
+
+* [shorter](https://www.w3.org/TR/css-color-4/#shorter)
+
+* [longer](https://www.w3.org/TR/css-color-4/#hue-longer)
+
+* [increasing](https://www.w3.org/TR/css-color-4/#increasing)
+
+* [decreasing](https://www.w3.org/TR/css-color-4/#decreasing)
+
+* [specified](https://www.w3.org/TR/css-color-4/#hue-specified)
+
+[hue-interpolation]: https://www.w3.org/TR/css-color-4/#hue-interpolation
+
+
+## Deprecated Functions
+
+Individual color-channel functions defined globally or in the color module are
+deprecated in favor of the new `color.channel()` function. That includes:
+
+* `color.red()`/`red()`
+* `color.green()`/`green()`
+* `color.blue()`/`blue()`
+* `color.hue()`/`hue()`
+* `color.saturation()`/`saturation()`
+* `color.lightness()`/`lightness()`
+* `color.whiteness()`
+* `color.blackness()`
+
+While deprecated, if the specified color argument is not a [legacy color][],
+throw an error.
+
+
+## New Color Module Functions
+
+These new functions are part of the built-in `sass:color` module.
+
+### `color.space()`
+
+* ```
+  space($color)
+  ```
+
+  * If `$color` is not a color, throw an error.
+
+  * Return a quoted string with the name of `$color`s associated
+    [color space][].
+
+### `color.to-space()`
+
+* ```
+  to-space($color, $space)
+  ```
+
+  * If `$color` is not a color, throw an error.
+
+  * Let `origin-space` be the result of calling `space($color)`.
+
+  * If `origin-space` equals-equals `$space`, return `$color`.
+
+    > This allows unknown spaces, as long as they match the origin space.
+
+  * If `$space` is not a [color space][], throw an error.
+
+  * Return the result of [converting][] the `origin-color`
+    `$color` to the `target-space` `$space`.
+
+### `color.is-legacy()`
+
+* ```
+  is-legacy($color)
+  ```
+
+  * If `$color` is not a color, throw an error.
+
+  * Return `true` if `$color` is a [legacy color][], or `false`
+    otherwise.
+
+### `color.is-powerless()`
+
+* ```
+  is-powerless($color, $channel, $space)
+  ```
+
+  * If `$color` is not a valid color, throw an error.
+
+  * If `$space` is null:
+
+    * Let `color` be the value of `color`, and let `space` be the result of
+      calling `space($color)`.
+
+  * Otherwise:
+
+    * Let `color` be the result of calling `to-space($color, $space)`, and let
+      `space` be the value of `$space`.
+
+  * If `$channel` is not the name of a channel in the color-space `space`,
```

Comment:
This should probably also include an error if `space` is not a known color space.